integration/pvtdata/data_purge_test.go

```diff
@@ -162,6 +162,22 @@ var _ = Describe("Pvtdata purge", func() {
 		//    - Repeat above to purge all keys to test the corner case
 		PIt("should remove all purged data from an eligible peer")
 
+		It("should purge data that has already been deleted", func() {
+			marblechaincodeutil.AssertPresentInCollectionM(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
+			marblechaincodeutil.AssertPresentInCollectionMPD(network, channelID, chaincode.Name, "test-marble-0", org2Peer0)
+
+			marblechaincodeutil.DeleteMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-0"}`, org2Peer0)
+
+			marblechaincodeutil.AssertDoesNotExistInCollectionM(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
+			marblechaincodeutil.AssertDoesNotExistInCollectionMPD(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
+
+			marblechaincodeutil.PurgeMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-0", "color":"blue"}`, org2Peer0)
+
+			// TODO this doesn't actually verify that the data has been purged which seems problematic
```

Comment:
Having thought about this a bit more, I think this is a hint that there's something missing- see #3844